src/ui/src/builder/panels/BuilderBlueprintLibraryPanel.vue

```diff
@@ -0,0 +1,190 @@
+<template>
+	<WdsModal
+		v-if="isOpen && isBlueprintLibraryEnabled"
+		title="Blueprint Library"
+		size="wide"
+		display-close-button
+		@close="handleClose"
+	>
+		<div class="BuilderBlueprintLibraryPanel">
+			<div class="BuilderBlueprintLibraryPanel__header">
+				<div class="BuilderBlueprintLibraryPanel__search">
+					<WdsTextInput
+						v-model="searchQuery"
+						placeholder="Search blueprints..."
+						left-icon="search"
+					/>
+				</div>
+			</div>
+
+			<div class="BuilderBlueprintLibraryPanel__content">
+				<div
+					v-if="isLoading"
+					class="BuilderBlueprintLibraryPanel__loading"
+				>
+					<p>Loading blueprints...</p>
+				</div>
+
+				<div
+					v-else-if="blueprints.length === 0"
+					class="BuilderBlueprintLibraryPanel__empty"
+				>
+					<p>No blueprints found</p>
+					<p
+						v-if="searchQuery"
+						class="BuilderBlueprintLibraryPanel__empty__hint"
+					>
+						Try adjusting your search
+					</p>
+				</div>
+
+				<div v-else class="BuilderBlueprintLibraryPanel__grid">
+					<BuilderBlueprintLibraryItem
+						v-for="blueprint in blueprints"
+						:key="blueprint.id"
+						:block="blueprint"
+					/>
+				</div>
+			</div>
+		</div>
+	</WdsModal>
+</template>
+
+<script setup lang="ts">
+import { ref, computed, watch, inject } from "vue";
+import WdsModal from "@/wds/WdsModal.vue";
+import BuilderBlueprintLibraryItem from "./BuilderBlueprintLibraryItem.vue";
+import WdsTextInput from "@/wds/WdsTextInput.vue";
+import injectionKeys from "@/injectionKeys";
+import { useToasts } from "../useToast";
+import { useDebouncer } from "@/composables/useDebouncer";
+
+const props = defineProps<{
+	modelValue: boolean;
+}>();
+
+const emit = defineEmits<{
+	"update:modelValue": [value: boolean];
+}>();
+
+const wf = inject(injectionKeys.core);
+const { pushToast } = useToasts();
+
+const isOpen = computed({
+	get: () => props.modelValue,
+	set: (value) => emit("update:modelValue", value),
+});
+
+const isBlueprintLibraryEnabled = computed(
+	() =>
+		Array.isArray(wf.featureFlags.value) &&
+		wf.featureFlags.value.includes("blueprint_library"),
+);
+
+const searchQuery = ref("");
+const blueprints = ref<
+	Array<{
+		id: string;
+		title: string;
+		description: string;
+		category: string;
+		version_number: number;
+	}>
+>([]);
+const isLoading = ref(false);
+
+async function loadBlueprints() {
+	if (!isBlueprintLibraryEnabled.value) return;
+
+	isLoading.value = true;
+	try {
+		const params = new URLSearchParams();
+		if (searchQuery.value) {
+			params.append("search", searchQuery.value);
+		}
+
+		const response = await fetch(
+			`/api/block-library/blocks?${params.toString()}`,
+		);
```

Comment:
You can abort the query
```suggestion
const abort = useAbortController():

async function loadBlueprints() {
	if (!isBlueprintLibraryEnabled.value) return;

	isLoading.value = true;
	try {
		const params = new URLSearchParams();
		if (searchQuery.value) {
			params.append("search", searchQuery.value);
		}

		const response = await fetch(
			`/api/block-library/blocks?${params.toString()}`,
			{ signal: abort.signal }
		);
```